flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/NFA.java

```diff
@@ -551,18 +579,47 @@ private boolean isSelfIgnore(final StateTransition<T> edge) {
 								nextVersion,
 								startTimestamp);
 					}
+
+					switch (skipStrategy.getStrategy()) {
+						case SKIP_PAST_LAST_ROW:
+							if (nextState.isFinal()) {
+								resultingComputationStates.add(createStartComputationState(computationState, outgoingEdges));
+							}
+							break;
+						case SKIP_TO_FIRST:
+							if (nextState.getName().equals(skipStrategy.getRpv()) &&
+								!nextState.getName().equals(currentState.getName())) {
+								ComputationState<T> startComputationState = createStartComputationState(computationState, outgoingEdges);
```

Comment:
The `outgoingEdges` parameter in this case will not work. Previously the assumption was that we created the new `start` state always in the starting state. So we could calculate the version for new run based on outgoing edges from the start state. 

In this case the outgoingEdges do not correspond to the starting state.